action-server/src/listeners/dbListeners.ts

```diff
@@ -0,0 +1,163 @@
+import { readFile } from "node:fs/promises";
+import type http from "node:http";
+import * as path from "node:path";
+import type { Pool, PoolClient } from "pg";
+import { configuration } from "../config";
+import { ActionsDbManager } from "../db";
+import { ActionWorkerPool } from "../threads/workerPool";
+import type { ActionDefinitionInsertedPayload, ActionResponse, ActionRunInsertedPayload } from "../type/types";
+import { extractSchemas } from "../utils/codeRunner";
+
+let listenClient: PoolClient | undefined;
+
+async function readFileFromStore(fileName: string): Promise<string> {
+  // read file from aerie file store and return [resolve] it as a string
+  const fileStoreBasePath = configuration().ACTION_LOCAL_STORE;
+  const filePath = path.join(fileStoreBasePath, fileName);
+  console.log(`path is ${filePath}`);
+  return await readFile(filePath, "utf-8");
+}
+
+async function handleActionDefinition(payload: ActionDefinitionInsertedPayload) {
+  console.log("action definition inserted");
+
+  // read the action file and extract parameter/setting schemas
+  const actionJS = await readFileFromStore(payload.action_file_path);
+  // console.debug(actionJS);
+  const schemas = await extractSchemas(actionJS);
+
+  console.info(`schemas ${JSON.stringify(schemas, null, 2)}`);
+
+  // todo: set schemas on the DB row?
+  const pool = ActionsDbManager.getDb();
+  const query = `
+    UPDATE actions.action_definition
+    SET
+      parameter_schema = parameter_schema || $1::jsonb,
+      settings_schema = settings_schema || $2::jsonb
+    WHERE id = $3
+      RETURNING *;
+  `;
```

Comment:
@dandelany I believe we are setting the schema in the DB with the query below. Is this TODO needed?

Reply:
Better question: why is this query merging the schemas? When we update a schema, we _replace_ the old value with the new one. If you merge it, then you run into issues where, for example, a parameter is dropped, but that parameter is still in the schema. 

Please reference `CreateModelParametersAction.java` and `UpdateConstraintParametersAction.java` for examples.

Reply:
Good catch @Mythicaeda - updated this to replace instead of merging, thanks